File: #7 reverse integer/reverse_integer/src/reverse_integer.rs
pub fn reverse(x: i32) -> i32 {
    let sol: i32 = x.abs()
        .to_string()
        .chars()
        .rev()
        .collect::<String>()
        .parse()
        .unwrap_or(0);
    if x < 0 {-sol} else {sol}
}

File: #7 reverse integer/reverse_integer.py
def reverse(n: int) -> int:
    if n == 0:
        return 0
    ret = int(str(abs(n))[::-1]) * int(n / abs(n))
    return ret if abs(ret) < (1<<31)-1 else 0


print(reverse(-123))
print(reverse(1534236469))
print(1<<31)
